### ISSUE TITLE
打开enable_cluster_plan，update from where语句出现core

### ISSUE THREAD

**Issue by mass3** [ADBSQL/AntDB#31]
打开enable_cluster_plan，update from where语句出现core
```
postgres=# CREATE TABLE writetest (a int);
CREATE TABLE
postgres=# CREATE TEMPORARY TABLE temptest (a int);
CREATE TABLE
postgres=#   
postgres=# UPDATE temptest SET a = 0 FROM writetest WHERE temptest.a = 1 AND writetest.a = temptest.a; 
ERROR:  fail to fetch slot from self reduce
DETAIL:  the peer of [REDUCE 11990] has performed an orderly shutdown
ERROR:  error message from poolmgr:reconnect three thimes , FATAL:  the database system is in recovery mode

ERROR:  error message from poolmgr:reconnect three thimes , FATAL:  the database system is in recovery mode

ERROR:  error message from poolmgr:reconnect three thimes , FATAL:  the database system is in recovery mode

ERROR:  error message from poolmgr:reconnect three thimes , FATAL:  the database system is in recovery mode

PANIC:  ERRORDATA_STACK_SIZE exceeded
server closed the connection unexpectedly
        This probably means the server terminated abnormally
        before or while processing the request.
The connection to the server was lost. Attempting reset: Failed.
!> 
```

测试sql：
```
CREATE TABLE writetest (a int);
CREATE TEMPORARY TABLE temptest (a int);
UPDATE temptest SET a = 0 FROM writetest WHERE temptest.a = 1 AND writetest.a = temptest.a; 

```

**Comment by hsiaoby** [comment]
fixed by 850072921edc2f48129cf8217f106bbc761c28a3

**Comment by mass3** [comment]
临时表和非临时表join update，结果正确
```
postgres=# CREATE TABLE writetest (a int);
CREATE TEMPORARY TABLE temptest (a int);
UPDATE temptest SET a = 0 FROM writetest WHERE temptest.a = 1 AND writetest.a = temptest.a; 
CREATE TABLE
postgres=# CREATE TEMPORARY TABLE temptest (a int);
CREATE TABLE
postgres=# UPDATE temptest SET a = 0 FROM writetest WHERE temptest.a = 1 AND writetest.a = temptest.a; 
UPDATE 0
postgres=# 
```
